src/sentry/issues/endpoints/organization_group_search_views.py

```diff
@@ -110,6 +115,55 @@ def get(self, request: Request, organization: Organization) -> Response:
                     data={"detail": "You do not have access to any projects."},
                 )
 
+        visibility = request.GET.getlist("visibility")
+        valid_visibility_options = [v[0] for v in GroupSearchViewVisibility.as_choices()]
+        if visibility:
+            if any(v not in valid_visibility_options for v in visibility):
+                return Response(
+                    status=status.HTTP_400_BAD_REQUEST,
+                    data={"detail": "Invalid visibility query parameter."},
+                )
+
+            if len(visibility) > 1:
+                org_query = GroupSearchView.objects.filter(
+                    organization=organization,
+                    visibility=GroupSearchViewVisibility.ORGANIZATION,
+                ).prefetch_related("projects")
+
+                owner_query = GroupSearchView.objects.filter(
+                    organization=organization,
+                    user_id=request.user.id,
+                    visibility=GroupSearchViewVisibility.OWNER,
+                ).prefetch_related("projects")
+
+                query = org_query.union(owner_query)
+            elif visibility[0] == GroupSearchViewVisibility.ORGANIZATION:
+                query = GroupSearchView.objects.filter(
+                    organization=organization,
+                    visibility=GroupSearchViewVisibility.ORGANIZATION,
+                ).prefetch_related("projects")
+            else:
+                query = GroupSearchView.objects.filter(
+                    organization=organization,
+                    user_id=request.user.id,
+                    visibility=GroupSearchViewVisibility.OWNER,
+                ).prefetch_related("projects")
```

Comment:
something like this may be a bit easier to maintain
```suggestion
            from functools import reduce
            from operator import or_
            ...
            org_query = GroupSearchView.objects.filter(
                organization=organization,
                visibility=GroupSearchViewVisibility.ORGANIZATION,
            ).prefetch_related("projects")

            owner_query = GroupSearchView.objects.filter(
                organization=organization,
                user_id=request.user.id,
                visibility=GroupSearchViewVisibility.OWNER,
            ).prefetch_related("projects")
            param_query_map = {
                  'organization':org_query
                  'owner': owner_query 
            }
            query_list = [param_query_map[v] for v in visibility]
            query = reduce(or_, query_list)
```

Reply:
This is way better, thanks for writing this out!